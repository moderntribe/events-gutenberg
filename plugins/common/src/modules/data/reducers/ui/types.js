/**
 * Internal dependencies
 */
import { PREFIX_COMMON_STORE } from '@moderntribe/common/data/utils';

export const OPEN_ACCORDION = `${ PREFIX_COMMON_STORE }/OPEN_ACCORDION`;
export const CLOSE_ACCORDION = `${ PREFIX_COMMON_STORE }/CLOSE_ACCORDION`;
export const ADD_ACCORDION = `${ PREFIX_COMMON_STORE }/ADD_ACCORDION`;
export const REMOVE_ACCORDION = `${ PREFIX_COMMON_STORE }/REMOVE_ACCORDION`;
