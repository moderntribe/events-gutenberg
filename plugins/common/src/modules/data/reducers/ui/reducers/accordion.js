/**
 * Internal dependencies
 */
import * as types from '../types';

export default ( state = {}, action ) => {
	switch ( action.type ) {
		case types.TOGGLE_ACCORDION:
			return {
				...state,
				[ action.payload.accordionId ]: ! state[ action.payload.accordionId ],
			};
		case types.ADD_ACCORDION:
			return {
				...state,
				[ action.payload.accordionId ]: false,
			};
		case types.REMOVE_ACCORDION:
			const { [ action.payload.accordionId ]: value, ...filteredState } = state;
			return filteredState;
		default:
			return state;
	}
};
