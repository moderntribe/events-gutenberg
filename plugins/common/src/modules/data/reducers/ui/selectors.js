/**
 * External dependencies
 */
import { createSelector } from 'reselect';

export const getUI = ( state ) => state.ui;

export const getAccordionId = ( state, ownProps ) => ownProps.accordionId;

export const getAccordions = createSelector(
	[ getUI ],
	( ui ) => ui.accordion,
);

export const getAccordionState = createSelector(
	[ getAccordions, getAccordionId ],
	( accordions, accordionId ) => accordions[ accordionId ],
);
