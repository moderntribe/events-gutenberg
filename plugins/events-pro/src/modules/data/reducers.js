/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import blocks from './blocks';

export default combineReducers( { blocks } );
