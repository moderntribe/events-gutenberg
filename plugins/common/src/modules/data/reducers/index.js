/**
 * External dependencies
 */
import { setupCreateReducer } from '@nfen/redux-reducer-injector';

/**
 * Internal dependencies
 */
import plugins from './plugins';
import ui from './ui';

export default setupCreateReducer( {
	plugins,
	ui,
} );
