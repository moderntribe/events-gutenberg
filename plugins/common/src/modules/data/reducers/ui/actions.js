/**
 * Internal dependencies
 */
import { types } from '@moderntribe/common/data/reducers/ui';

export const addAccordion = ( accordionId ) => ( {
	type: types.ADD_ACCORDION,
	payload: {
		accordionId,
	},
} );

export const removeAccordion = ( accordionId ) => ( {
	type: types.REMOVE_ACCORDION,
	payload: {
		accordionId,
	},
} );

export const toggleAccordion = ( accordionId ) => ( {
	type: types.TOGGLE_ACCORDION,
	payload: {
		accordionId,
	},
} );
