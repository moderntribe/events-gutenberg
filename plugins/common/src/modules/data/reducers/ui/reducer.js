/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import { accordion } from './reducers';

export default combineReducers( {
	accordion,
} );
