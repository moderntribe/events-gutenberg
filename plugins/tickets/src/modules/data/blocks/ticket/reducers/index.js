/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import ui from './ui';
import tickets from './tickets';

export default combineReducers( {
	ui,
	tickets,
} );
