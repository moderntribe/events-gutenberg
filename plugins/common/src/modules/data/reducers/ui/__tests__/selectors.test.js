/**
 * Internal dependencies
 */
import { selectors } from '@moderntribe/common/data/reducers/ui';

const state = {
	ui: {
		accordion: {
			id1: false,
		}
	}
};

const ownProps = { accordionId: 'id1' };

describe( 'UI Selectors', () => {
	it( 'should return the UI state', () => {
		expect( selectors.getUI( state ) ).toMatchSnapshot();
	} );

	it( 'should return the accordion id', () => {
		expect( selectors.getAccordionId( state, ownProps ) ).toMatchSnapshot();
	} );

	it( 'should return the accordion state', () => {
		expect( selectors.getAccordions( state ) ).toMatchSnapshot();
	} );

	it( 'should return the accordion state of accordion id', () => {
		expect( selectors.getAccordionState( state, ownProps ) ).toMatchSnapshot();
	} );
} )
