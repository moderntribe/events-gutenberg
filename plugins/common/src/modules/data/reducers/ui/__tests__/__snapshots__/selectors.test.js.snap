// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Selectors should return the UI state 1`] = `
Object {
  "accordion": Object {
    "id1": false,
  },
}
`;

exports[`UI Selectors should return the accordion id 1`] = `"id1"`;

exports[`UI Selectors should return the accordion state 1`] = `
Object {
  "id1": false,
}
`;

exports[`UI Selectors should return the accordion state of accordion id 1`] = `false`;
